modules/rpc/yarpc.go

```diff
@@ -128,20 +128,23 @@ func (m *YarpcModule) Start(readyCh chan<- struct{}) <-chan error {
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
-	channel, err := tchannel.NewChannel(m.config.AdvertiseName, nil)
+	interceptor := yarpc.UnaryInboundMiddleware(m.inboundMiddlewares...)
+	onewayInterceptor := yarpc.OnewayInboundMiddleware(m.onewayInboundMiddlewares...)
+
+	tchTransport, err := tch.NewChannelTransport(
+		tch.ServiceName(m.config.AdvertiseName),
+		tch.ListenAddr(m.config.Bind),
+	)
 	ret := make(chan error, 1)
```

Comment:
unrelated nit (i think this is my code anyway): move this above line 134 so the err check is right after we potentially get the error